server/Dockerfile

```diff
@@ -27,4 +27,10 @@ COPY . .
 
 COPY --from=development /usr/src/app/dist ./dist
 
-CMD ["node", "dist/main"] 
\ No newline at end of file
+CMD ["node", "dist/main"] 
+
+FROM chronicle-api:1.0.0 as seed
+
+COPY package.json ./
+
+COPY prisma/* ./prisma/
```

Comment:
Not sure about having this in the same dockerfile as the rest of the API endpoint. You should move it to a separate docker file called `Seed.Dockerfile` then pass `Seed.Dockerfile` to docker compose.

Since prisma is the only dependency, you can just use a node image, install prisma globally then copy the files over and run prisma generate immediately. It is fine for our case since the image will only have one project anyways.

```Dockerfile
FROM node:16
RUN yarn global add prisma@3.10.0
COPY package.json ./
COPY prisma/* ./prisma/

# Run db command
CMD prisma db seed
```

Then in the compose file just do

```yml
db_seed:
    build: 
      context: ./server
      dockerfile: Seed.Dockerfile
    depends_on:
      - chronicle-database
      - mongo_replica
```

Reply:
the reason why I didn't use `node:16` was it was giving me an issue with ts (I know this is the case cause I added a step to install ts-node and it worked fine afterwards)

I can update it to sit in a separate dockerfile + add prisma as a dep.